### DISCUSSION TITLE
Maplibre ios tileserveroptions

### DISCUSSION THREAD

**Discussion by MananJ-IOS** [maplibre/maplibre-native#290] [General]
Hi, can anyone help me out on how to use the MGLTileServerOptions?. I need to change the baseUrl along with apiKeyParameterName both of which are offered through tile server options. Although, I am unable to update them using MGLSettings.

**Comment by wipfli** [comment]
We don't have a documentation website yet for native, but you might find some info here: 

* https://docs.maptiler.com/maplibre-gl-native-android/
* https://github.com/aws-samples/amazon-location-samples

**Reply by MananJ-IOS** [reply]
Thanks for your reply. I realize that there is no documentation regarding this but I need to know if hosting custom tiles is possible without rebuilding the source code. Also, how do I utilize TileServerOptions ?. I have all the required custom URLs (tiles, glyphs, sprites), all I need is to replace them but that does not seem to be working.

**Reply by wipfli** [reply]
We published the android API now at https://maplibre.org/maplibre-gl-native/android/api/. Probably this does not solve your problem though, @MananJ-IOS ...

**Reply by wipfli** [reply]
Would be so cool to have a few working examples somewhere which would demonstrate how one can do different things.

**Reply by wipfli** [reply]
Maybe this example from @roblabs could help? https://github.com/roblabs/openmaptiles-ios-demo

**Reply by wipfli** [reply]
Maybe this stuff helps? https://github.com/maplibre/maplibre-gl-native/issues/17
